samples/SampleMvvmApp/Services/NotifictionService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Notifications;
+using Avalonia.Threading;
+
+namespace SampleMvvmApp.Services
+{
+    public class NotificationService : INotificationService
+    {
+        private int _notificationTimeout = 10;
+        private WindowNotificationManager? _notificationManager;
+
+        public int NotificationTimeout
+        {
+            get => _notificationTimeout;
+            set
+            {
+                _notificationTimeout = (value < 0) ? 0 : value;
+            }
+        }
+
+        /// <summary>Set the host window.</summary>
+        /// <param name="hostWindow">Parent window.</param>
+        public void SetHostWindow(Window hostWindow)
+        {
+            var notificationManager = new WindowNotificationManager(hostWindow)
+            {
+                Position = NotificationPosition.BottomRight,
+                MaxItems = 4,
+                Margin = new Thickness(0, 0, 15, 40)
+            };
+
+            _notificationManager = notificationManager;
+        }
+
+        /// <summary>Display the notification.</summary>
+        /// <param name="title">Title.</param>
+        /// <param name="message">Message.</param>
+        /// <param name="onClick">Optional OnClick action.</param>
+        public void Show(string title, string message, Action? onClick = null)
+        {
+            if (_notificationManager is { } nm)
+            {
+                // Alt method:
+                //// RxApp.MainThreadScheduler.Schedule(() =>
+                Dispatcher.UIThread.InvokeAsync(() =>
```

Comment:
@DamianSuess Why ThreadSafe should be check in the service layer? Is it call by app directly? then should be manage by app

Reply:
Good call 👍

Reply:
@OmidID thanks again for taking the time. In this instance, we didn't need the UI Thread call anyways.

Just a heads up, I'm gonna have a couple more PRs coming up in preparation for the Avalonia 0.11 releases